contributors/devel/sig-architecture/conformance-tests.md

```diff
@@ -17,29 +17,30 @@ NB: This should be viewed as a living document in a few key areas:
   current set of e2e tests, as such this document is currently intended to
   guide us in the addition of new e2e tests than can fill this gap
 - This document currently focuses solely on the requirements for GA,
-  non-optional features or APIs. The list of requirements will be refined over
-  time to the point where it as concrete and complete as possible.
-- There are currently conformance tests that violate some of the requirements
-  (e.g., require privileged access), we will be categorizing these tests and
-  deciding what to do once we have a better understanding of the situation
-- Once we resolve the above issues, we plan on identifying the appropriate areas
-  to relax requirements to allow for the concept of conformance Profiles that
-  cover optional or additional behaviors
+  non-optional features or APIs. Upcoming profiles will cover optional features
+  and require some modifications to these rules.
 
-## Conformance Test Requirements
+## Conformance Profiles
 
-Conformance tests currently test only GA, non-optional features or APIs. More
-specifically, a test is eligible for promotion to conformance if:
+Conformance tests are grouped into [profiles], which allow for testing
+conformance of optional features that every cluster need
+not provide. Profiles are _additive_, meaning that supporting one profile cannot
+"take away" tests from another profile.
 
-- it tests only GA, non-optional features or APIs (e.g., no alpha or beta
-  endpoints, no feature flags required, no deprecated features)
+Starting in 1.19, there are two profiles defined:
+Base and Privileged. All conforming Kubernetes clusters must meet the Base
+profile, but the Privileged profile is optional. Most tests in versions prior to
```

Comment:
> the Privileged profile is optional

would this move the extensibility mechanisms (webhooks and CRDs) to an optional profile? that seems like a step backward from a user perspective in having a consistent surface area to write to

Reply:
It would mean that clusters that allow those extensibility mechanisms must all behave in the same way, but that not all clusters must allow those. Again the goal here is trying to enable, for example, a shared or highly managed cluster to be conformant.

One option could be to differentiate between the "workload point of view" and the "cluster operator" point of view, and make separate profiles for them. We may do that in the future but this is intended as a first step.

Reply:
What @liggitt  mentions is the key concern of having profiles.  If you don't associate profiles with a process that is asking the already certified distributions which groups of the profiles they are able to support, and asks the distributions to drive towards all supporting the same groups of profiles,  you end up with the different distributions just supporting whatever profiles they want.  And the end result is that different Kubernetes distributions are perceived by users to not be interoperable and workloads become not portable.   This is not in the spirit of how we drove this work in March 2017 when it was initiated.  The best case for profiles would be that a large number of Kubernetes distributions would fit into the same equivalence class (e.g. the cloud provider profile group) and all those providers agree to support the same set of profiles.  And then maybe you have a lighter weight equivalence class of Kubernetes distributions that all agree to support the same smaller subset of profiles.  If you end up supporting more than two equivalence classes of profile groups the user just perceives Kubernetes as no longer interoperable.

Reply:
We definitely want to keep the number of profiles low, and we need a Base profile that covers the core functions for "ordinary" workloads that do not require special privileges. That's really what this is intended to cover for now.

Reply:
> It would mean that clusters that allow those extensibility mechanisms must all behave in the same way, but that not all clusters must allow those

Hmm... that seems more oriented at permissions of the user running the conformance job than the capabilities of the distribution. I'm not sure how that is different/better than the situation we have today which doesn't require multiple profiles. To certify conformance, a distribution must demonstrate that their server can have these tests run against it. It does not mean that all users are authorized to run a conformance job against all installations of the distribution. For example, Openshift is conformant, but users of its free online service are not allowed to install CRDs.

> we need a Base profile that covers the core functions for "ordinary" workloads that do not require special privileges

Is the goal of this separation for the benefit of manifest authors who can limit their manifests to a set of APIs and label the manifest as "Base profile conformant"?